openproblems/data/tabula_muris_senis.py

```diff
@@ -1,62 +1,47 @@
 import tempfile
 
+COLLECTION_ID = "0b9d8a04-bb9d-44da-aa27-705bb65b54eb"
+DOMAIN = "cellxgene.cziscience.com"
+API_BASE = f"https://api.{DOMAIN}"
+METHOD_ALIASES = {"10x 3' v2": "droplet", "Smart-seq2": "facs"}
 
-def get_filenames_and_urls(url_df, method_list=None, organ_list=None):
-    """Takes in dataframe and returns corresponding filename(s) and url(s).
 
-    Takes in dataframe (with sample information stored), a list of methods,
-    and a list of organs.
-    Returns filenames and figshare URLs associated with inputs.
-    If method_list or organ_list are None, do not filter based on that argument.
-    """
-    subset_df = url_df.copy()
-    # If method_list specified, filter based on methods in list.
-    if method_list:
-        subset_df = subset_df.loc[subset_df.method.isin(method_list)]
-    # If organ_list specified, filter based on organs in list.
-    if organ_list:
-        subset_df = subset_df.loc[subset_df.organ.isin(organ_list)]
+def matching_dataset(dataset, method_list, organ_list):
+    # we want to skip combined datasets
+    if len(dataset["tissue"]) > 1:
+        return False
+    if dataset["tissue"][0]["label"] not in organ_list:
+        return False
+    method = METHOD_ALIASES[dataset["assay"][0]["label"]]
+    return method in method_list
 
-    return subset_df
 
+def load_raw_counts(dataset):
+    dataset_id = dataset["id"]
+    assets_path = (
+        f"/curation/v1/collections/{COLLECTION_ID}/datasets/{dataset_id}/assets"
+    )
+    url = f"{API_BASE}{assets_path}"
+    res = requests.get(url=url)
+    assets = res.json()
+    assets = [asset for asset in assets if asset["filetype"] == "H5AD"]
+    asset = assets[0]
 
-def make_anndata_from_filename_and_url(filename, url, test=False):
-    """Takes in filename and url pair. Returns corresponding anndata object."""
+    filename = f"{COLLECTION_ID}_{dataset_id}_{asset['filename']}"
     with tempfile.TemporaryDirectory() as tempdir:
         filepath = os.path.join(tempdir, filename)
-        scprep.io.download.download_url(url, filepath)
+        scprep.io.download.download_url(asset["presigned_url"], filepath)
         adata = sc.read_h5ad(filepath)
-        utils.filter_genes_cells(adata)
 
-    if test:
-        sc.pp.subsample(adata, n_obs=100)
-        adata = adata[:, :1000]
-        utils.filter_genes_cells(adata)
-
-    return adata
-
-
-def make_anndata_list(subset_df, test):
-    """Makes anndata from filename/url pair. Adds to list of anndatas.
-
-    Input dataframe that contains filenames and urls to make anndatas from.
-    Returns a list of anndata objects.
-    """
-    adata_list = []
-    for i in range(len(subset_df)):
-        row = subset_df.iloc[i]
-        adata_list.append(
-            make_anndata_from_filename_and_url(row.filename, row.figshare_url)
-        )
-    if test:
-        return adata_list[0]
-    return adata_list
+    utils.filter_genes_cells(adata)
+    # raw counts are in `raw`
+    return adata.raw.to_adata()
```

Comment:
Maybe add a note here saying that raw counts are only in raw if it exists, otherwise they are in `adata.X`.